[RFC] Log PT2 chromium events to scuba

jamesjwu opened the pull request:
Summary:
This diff implements a bunch of views for internal scuba viewing. 

TODOS that I might punt to another diff:
- Saving cache stats via counter is definitely sus here, but there's not really a good way to track "fx graph cache hit for this compile phase" right now. Will think about this more.
- We should definitely log frame id, compile id, etc 
- We should definitely be logging configs. That way, we can A/B test based on whether a config is turned on.
- idk what I'm doing with compile_uuid yet, but it's useful when you want to look at samples for a single run. I think if we had mast job info this field is not needed, but it's nice to be able to drill down to a single run and get its chrome trace view or icicle view, so idk

Test Plan:
All of the above views are run with nanogpt benchmark:

```
buck run mode/opt caffe2/benchmarks/dynamo:torchbench -- --training --backend=inductor --only nanogpt --performance
```

Differential Revision: D61392607




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @rec

ezyang commented on torch/_utils_internal.py:
```diff
@@ -237,3 +237,7 @@ def max_clock_rate():
+
+
+def log_chromium_event_internal(event, stack, compile_uuid, start_timestamp=None):
+    return None
```
Will be nice to get on the unified Logger API train soon

ezyang commented on torch/_dynamo/utils.py:
```diff
@@ -251,16 +261,20 @@ def dynamo_timed(
     fail_reason: Optional[str] = None
     time_spent = float("-inf")
+    if phase_name == "entire_frame_compile":
+        reset_chromium_events()
```
What's going on here? (Is it required to hardcode this specific phase name?)

jamesjwu replied:
Honestly we don't even need this, I had it before when I wanted to reset the uuid every compile. Gonna remove

ezyang commented on torch/_dynamo/utils.py:
```diff
@@ -807,6 +821,18 @@ def get_compilation_metrics() -> List[Union[CompilationMetrics, BwdCompilationMe
 
+chromium_event_stack = ["__start__"]
+# Generate a unique id for this process, which we can use in scuba to filter down
+# to a single python run.
```
It's weird that there's not already a canonical field or something like for this kind of use case

ezyang commented on torch/_dynamo/utils.py:
```diff
@@ -807,6 +821,18 @@ def get_compilation_metrics() -> List[Union[CompilationMetrics, BwdCompilationMe
+
+def reset_chromium_events() -> None:
+    global chromium_event_stack
+    chromium_event_stack = ["__start__"]
```
Do you want to keep reallocating this list, or maybe .clear().push() is good?

ezyang commented on torch/_dynamo/utils.py:
```diff
@@ -826,18 +852,22 @@ def log_event_start(
         :param metadata: Any extra metadata associated with this event
         """
-        ChromiumEventLogger._log_timed_event(
+        global chromium_event_stack
```
nit: you don't have to global it here, because you don't assign over it

ezyang commented on torch/_dynamo/utils.py:
```diff
@@ -807,6 +821,18 @@ def get_compilation_metrics() -> List[Union[CompilationMetrics, BwdCompilationMe
     return list(_compilation_metrics)
 
 
+chromium_event_stack = ["__start__"]
```
nit: all caps because it's global

ezyang reviewed (approved):
okey dokey

jamesjwu committed:
[RFC] Log PT2 chromium events to scuba (#133859)

Summary:
X-link: https://github.com/pytorch/benchmark/pull/2424

Pull Request resolved: https://github.com/pytorch/pytorch/pull/133859

This diff implements a bunch of views for internal scuba viewing.

TODOS that I might punt to another diff:
- Saving cache stats via counter is definitely sus here, but there's not really a good way to track "fx graph cache hit for this compile phase" right now. Will think about this more.
- We should definitely log frame id, compile id, etc
- We should definitely be logging configs. That way, we can A/B test based on whether a config is turned on.
- idk what I'm doing with compile_uuid yet, but it's useful when you want to look at samples for a single run. I think if we had mast job info this field is not needed, but it's nice to be able to drill down to a single run and get its chrome trace view or icicle view, so idk

Test Plan:
All of the above views are run with nanogpt benchmark:

```
buck run mode/opt caffe2/benchmarks/dynamo:torchbench -- --training --backend=inductor --only nanogpt --performance
```

Reviewed By: ezyang

Differential Revision: D61392607
